### PR TITLE
BL-72-fe-as-a-user-i-can-filter-asylum-case-decisions-by-one-or-more-columns-from-the-dataset

### PR DESCRIPTION
**Pull request by VWashingtonCoder**
## Contributors
Vince Andre Washington (Author)
Ryan Magro (Co-Author)

## Description
Bl-72 FE: As a user, I can filter asylum case decisions by one or more columns from the dataset.

* Built in advanced search filter which allows for advanced search with multiple columns at once. 

#### Video Link

[Loom Video](https://www.loom.com/share/fda1ebb63c704f22b12b6293c51f02ce)

#### Jira Link

https://bloomtechlabs.atlassian.net/browse/BL-72?atlOrigin=eyJpIjoiM2E4Yjc5YTgxZWM5NDY1ZDg4MDRjNWRiNTA5MGI4ZDQiLCJwIjoiaiJ9

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by Mbravo36 — COMMENTED** [review]
Overall, I think the code was well put together and its very easy for the user to be able to find what they are looking for. The search bar is easy to use and the table works well with it the bring the desired data on to it.

**Review by SafiaOmar — DISMISSED** [review]

**Review by vbrambila2 — DISMISSED** [review]
I liked the idea of the checkboxes. As a user, I would find it more simple to click on the box than to type into the search box

**Review by XavierTwitty — APPROVED** [review]
you went above and beyond with the advanced search. I like how you made a showAdvancedSearch function to hide or show the component when clicked on.

**Review by EricMarshallLambert — APPROVED** [review]
This is a good testing mechanism, and allows backend to inspect and potentially reuse JS logic when implementing in the API.

**Review by Ricardorouge — APPROVED** [review]
very well done, the implementation of the advanced search looks amazing

File: src/components/pages/Table/SearchBar/AdvancedSearch.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      {/* <Form.Item label="Race/Ethnicity" name="raceOrEthnicity">
+        <Checkbox.Group options={raceOrEthnicityOptions} />
+      </Form.Item> */}
+
+      <Form.Item label="Case Outcome" name="caseOutcome">
+        <Checkbox.Group options={caseOutcomeOptions} />
+      </Form.Item>
+
+      <Form.Item label="Completion Date" name="completionDate">
+        <RangePicker />
```
Comment (Joshmclemore) [inline src/components/pages/Table/SearchBar/AdvancedSearch.js:89]: Great job on the RangePicker! I love this implementation

**Review by Joshmclemore — DISMISSED** [review]

**Review by Joshmclemore — APPROVED** [review]
Thanks for your work updating the state management and resolving all of the merge conflicts guys! It looks great and is hopefully conflict free now.

**Review by MaxwellChrist — APPROVED** [review]
Vince and Ryan, this looks amazing and you both did an incredible job. I really like the advanced search almost as a default option instead of the search bar after seeing the loom video. Vince was explaining that the search bar can be combined with other search customizations, like selecting Algerian citizenship while keeping some of the boxes for other search options checked in the advanced search bar. His comment "I'm not sure why a user wouldn't click it" sums up my feeling about the search bar at the moment. Especially as users might misspell things, then we have to discuss if/where we enter error messages, maybe even pre-filled options in the search menu, etc. There can be other tickets and discussions to see if the search bar is really necessary, but the fact that you have everything functioning properly is very impressive

**Pull request merged**